WizardEngine/Entity.cpp

```diff
@@ -30,78 +44,85 @@ bool Entity::Update(float deltaTime) {
 		components[i]->Update(deltaTime);
 	return true;
 }
-
-void Entity::SetRotation(DirectX::XMFLOAT3 a_vRotation) 
+#pragma region Accessors
+XMFLOAT3 Entity::GetPosition()
 {
-	m_vRotation = a_vRotation;
+	return m_vPos;
 }
 
-void Entity::SetScale(DirectX::XMFLOAT3 a_vScale)
+XMFLOAT3 Entity::GetRotation()
 {
-	m_vScale = a_vScale;
+	return m_vRotation;
 }
 
-void Entity::SetPos(DirectX::XMFLOAT3 a_vPos)
+XMFLOAT3 Entity::GetScale()
 {
-	m_vPos = a_vPos;
+	return m_vScale;
 }
 
-void Entity::SetWorld(DirectX::XMFLOAT4X4 a_mWorld)
+XMFLOAT4X4 Entity::GetWorldMatrix()
 {
-	m_mWorld = a_mWorld;
+	if (outdatedMatrix) {
```

Comment:
using the dirty flag so the matrix isn't recalced when not needed. Also, nothing outside of the entity needs to call UpdateWorldView since it gets updated when needed.